aip/auth/4119.md

```diff
@@ -0,0 +1,142 @@
+---
+id: 4119
+scope: auth
+state: draft
+created: 2022-09-09
+---
+
+# mTLS Token Binding
+
+Token binding allows issuing Google access tokens that are bound to mTLS
+credentials. The advantage of such mTLS bound tokens is that they are meant to
+only be used over secure channels established via mTLS credentials they are
+bound to. Therefore, using bound tokens is more secure than bearer tokens which
+can be stolen and adversarially replayed.
+
+This AIP describes the flow of (1) obtaining access tokens bound to X.509
+certificate identities, called identity-bound tokens and (2) how to use them to
+access Google APIs using the Google auth libraries.
+
+**Note:** Because this AIP describes guidance and requirements in a
+language-neutral way, it uses generic terminology which may be imprecise or
+inappropriate in certain languages or environments.
+
+## Guidance
+
+If users enable token binding, they **should** do so via [ADC][0]. This section
+describes the general guidance of supporting such tokens.
+
+### Prerequisites
+
+Identity-bound access tokens require that the clients have
+[X.509 SPIFFE Verifiable Identity Documents][1] (SVIDs). [Mutual Authentication
+Using Workload Credentials][2] describes how such SVIDs are provisioned in
+Google Cloud.
+
+Additionally, Additionally, identity-bound access tokens tokens require
+configuring a workload identity pool and identity provider with Google Cloud's
+IAM. The instructions on how to do this are out of scope of this AIP.
+
+### Using mTLS Token Binding
+
+The auth libraries **must** support the following values in the
+**"~/.config/gcloud/certificate_config.json"** configuration file. Note that the
+default location of this file can be changed using the
+`GOOGLE_API_CERTIFICATE_CONFIG` environment variable.
+
+```json
+{
+  “version”: 1
+  "cert_info": {
```

Comment:
update the config schema here too. (I wonder if this can be consolidated into a single doc location in the future.)

Reply:
Fixed.

And good call. We can probably update the AIPs in the future and consolidate these schemas.